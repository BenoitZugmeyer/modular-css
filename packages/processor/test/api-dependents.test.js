"use strict";

const namer = require("@modular-css/test-utils/namer.js");
const relative = require("@modular-css/test-utils/relative.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe(".dependents()", () => {
        let processor;

        beforeEach(() => {
            processor = new Processor({
                namer,
            });
        });

        it("should return the dependents of the specified file", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(
                relative(processor.dependents(require.resolve("./specimens/local.css")))
            )
            .toMatchSnapshot();
        });

        it("should throw if no file is passed", async () => {
            await processor.file("./packages/processor/test/specimens/start.css");

            expect(() => processor.dependents()).toThrowErrorMatchingSnapshot();
        });
    });
});
