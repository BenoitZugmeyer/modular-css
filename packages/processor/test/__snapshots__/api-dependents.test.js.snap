// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js .dependents() should return the dependents of the specified file 1`] = `
Array [
  "packages/processor/test/specimens/start.css",
]
`;

exports[`/processor.js .dependents() should throw if no file is passed 1`] = `"Must provide a file to processor.dependants()"`;
